utilities/trace_analyzer_tool.cc

```diff
@@ -0,0 +1,1751 @@
+DEFINE_bool(print_key_distribution, false, "Print the key size distribution.");
+DEFINE_bool(output_value_distribution, false,
+            "Print the value size distribution, only available for Put.");
+DEFINE_int32(print_top_k_access, 1,
+             "<top K of the variables to be printed>"
+             "Print the top k accessed keys, top k accessed prefix"
+             "and etc.");
+DEFINE_int32(output_ignore_count, 0,
+             "<threshold>, ignores the access count <= this value,"
+             "it will shorter the output.");
+DEFINE_int32(value_interval, 8,
+             "To output the value distribution, we need to set the value"
+             "intervals andmake the statistic of the value size distribution"
+             "in different intervals. The default is 8.");
+
+namespace rocksdb {
+
+std::map<std::string, int> taOptToIndex = {
+    {"get", 0},          {"put", 1},   {"delete", 2},  {"single_delete", 3},
+    {"range_delete", 4}, {"merge", 5}, {"iterator", 6}};
+
+std::map<int, std::string> taIndexToOpt = {
+    {0, "get"},          {1, "put"},   {2, "delete"},  {3, "single_delete"},
+    {4, "range_delete"}, {5, "merge"}, {6, "iterator"}};
+
+// Transfer the Microsecond time to date time
+std::string TraceAnalyzer::MicrosdToDate(uint64_t time_in) {
+  time_t tx = static_cast<time_t>(time_in / 1000000);
+  int rest = static_cast<int>(time_in % 1000000);
+  std::string date_time(ctime(&tx));
+  date_time.pop_back();
+  date_time += " +: " + std::to_string(rest);
+  return date_time;
+}
+
+namespace {
+
+bool ReadOneLine(std::istringstream* iss, SequentialFile* seq_file,
+                 std::string* output, bool* has_data, Status* result) {
+  const int kBufferSize = 128;
+  char buffer[kBufferSize + 1];
+  Slice input_slice;
+
+  std::string line;
+  bool has_complete_line = false;
+  while (!has_complete_line) {
+    if (std::getline(*iss, line)) {
+      has_complete_line = !iss->eof();
+    } else {
+      has_complete_line = false;
+    }
+    if (!has_complete_line) {
+      // if we're not sure whether we have a complete line,
+      // further read from the file.
+      if (*has_data) {
+        *result = seq_file->Read(kBufferSize, &input_slice, buffer);
+      }
+      if (input_slice.size() == 0) {
+        // meaning we have read all the data
+        *has_data = false;
+        break;
+      } else {
+        iss->str(line + input_slice.ToString());
+        // reset the internal state of iss so that we can keep reading it.
+        iss->clear();
+        *has_data = (input_slice.size() == kBufferSize);
+        continue;
+      }
+    }
+  }
+  *output = line;
+  return *has_data || has_complete_line;
+}
+
+uint64_t MultiplyCheckOverflow(uint64_t op1, uint64_t op2) {
+  if (op1 == 0 || op2 == 0) {
+    return 0;
+  }
+  if (port::kMaxUint64 / op1 < op2) {
+    return op1;
+  }
+  return (op1 * op2);
+}
+
+void DecodeCFAndKey(std::string& buffer, uint32_t* cf_id, Slice* key) {
```

Comment:
I guess I can later pull these common functions to a separate utils file to remove some code redundancy.